test/JDBC/expected/Test-spatial-functions-vu-verify.out

```diff
@@ -0,0 +1,1211 @@
+SELECT * FROM EqualCTE WHERE Equal = 1.0 ORDER BY Equal;
+GO
+~~START~~
+int#!#bit
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+WITH EqualCTE AS ( SELECT ID, @referencePoint.STEquals(PointColumn) AS Equal FROM TestSpatialFunction_YourTableTemp ORDER BY PointColumn.STX)
+SELECT * FROM EqualCTE WHERE Equal != 1.0 ORDER BY Equal;
+GO
+~~START~~
+int#!#bit
+1#!#0
+2#!#0
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+DECLARE @isEqual BIT = 1;
+SELECT ID, PointColumn.STEquals(@referencePoint) AS EqualityReferencePoint,
+CASE WHEN PointColumn.STEquals(@referencePoint) = @isEqual THEN 'Close'
+ELSE 'Far'
+END AS Proximity
+FROM TestSpatialFunction_YourTableTemp;
+go
+~~START~~
+int#!#bit#!#text
+1#!#0#!#Far
+2#!#0#!#Far
+3#!#<NULL>#!#Far
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT ID, @referencePoint.STEquals(PointColumn) AS EqualityReferencePoint,
+CASE WHEN @referencePoint.STEquals(PointColumn) = @referencePoint.STY THEN 'Close'
+ELSE 'Far'
+END AS Proximity
+FROM TestSpatialFunction_YourTableTemp;
+go
+~~START~~
+int#!#bit#!#text
+1#!#0#!#Close
+2#!#0#!#Close
+3#!#<NULL>#!#Far
+~~END~~
+
+
+DECLARE @Ranges TABLE (MinDistance float, MaxDistance float);
+INSERT INTO @Ranges VALUES (0, 5), (5, 10), (10, 15);
+SELECT * FROM ( SELECT ID,
+CASE WHEN PointColumn1.STEquals(PointColumn2) BETWEEN 0 AND 1 THEN 'yes'
+ELSE 'no'
+END AS Range
+FROM TestSpatialFunction_YourTableTemp2
+) AS Source
+PIVOT ( COUNT(ID) FOR Range IN ([0-5], [5.1-10], [10.1-15], [15.1+])) AS PivotTable;
+go
+~~ROW COUNT: 3~~
+
+~~START~~
+int#!#int#!#int#!#int
+0#!#0#!#0#!#0
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT ID, PointColumn.STEquals(@referencePoint) AS Equality FROM <YourDatabase>.dbo.TestSpatialFunction_YourTableTemp;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: syntax error near '<' at line 2 and character position 66)~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT ID, PointColumn.STEquals(@referencePoint) AS Equal,
+JSON_QUERY('{"Equal":' + CAST(PointColumn.STEquals(@referencePoint) AS NVARCHAR(MAX)) + '}') AS Json
+FROM TestSpatialFunction_YourTableTemp;
+go
+~~START~~
+int#!#bit#!#nvarchar
+1#!#0#!#{"Equal": 0}
+2#!#0#!#{"Equal": 0}
+3#!#<NULL>#!#<NULL>
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT ID, @referencePoint.STEquals(PointColumn) AS Equal,
+JSON_QUERY('{"Equal":' + CAST(@referencePoint.STEquals(PointColumn) AS NVARCHAR(MAX)) + '}') Json
+FROM TestSpatialFunction_YourTableTemp;
+go
+~~START~~
+int#!#bit#!#nvarchar
+1#!#0#!#{"Equal": 0}
+2#!#0#!#{"Equal": 0}
+3#!#<NULL>#!#<NULL>
+~~END~~
+
+
+SELECT [PointColumn1].STEquals([PointColumn2]) AS Equality FROM [TestSpatialFunction_YourTableTemp2];
```

Comment:
Please use `ORDER BY` clause after each selects

Reply:
+1, this is a must have for each SELECT statement to make test output consistent, specially the test cases which outputs multiple rows.

Reply:
Resolved
Added order by clause with parameter SRid in slelect statements.

Reply:
Let's not add order by clause with SRID, rather let's use something like STX or LAT(in case of geography).

Reply:
resolved